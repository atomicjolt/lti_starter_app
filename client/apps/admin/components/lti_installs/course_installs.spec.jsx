import React from 'react';
import TestRenderer from 'react-test-renderer';
import CourseInstalls from './course_installs';

describe('lti installs course install', () => {
  let result;
  const courseId = 123;
  let instance;
  let changed;

  const applicationInstance = {
    name: 'application_name',
    lti_key: 'lti_key',
    lti_secret: 'lti_secret',
    lti_config_xml: 'lti_config_xml',
  };
  const canvasRequest = () => {};
  const onlyShowInstalledChanged = () => { changed = true; };
  const loadingCourses = {
    courseName: 'name',
    courseId: 'id',
  };
  const courses = [];
  const installedTool = {
    id: 12,
  };
  const courseName = 'courseName';

  beforeEach(() => {
    result = TestRenderer.create(<CourseInstalls
      applicationInstance={applicationInstance}
      canvasRequest={canvasRequest}
      onlyShowInstalledChanged={onlyShowInstalledChanged}
      loadingCourses={loadingCourses}
      courses={courses}
      installedTool={installedTool}
      courseName={courseName}
      courseId={courseId}
    />);
    instance = result.root;
  });

  it('renders', () => {
    expect(result).toBeDefined();
  });

  it('renders the form not null', () => {
    expect(result).not.toBeNull();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders buttons', () => {
    const courseNameSpan = instance.findByType('span');
    expect(courseNameSpan.props.children).toBe('Course Name');
  });

  // Remove the checkbox so removing the test for now
  // it('handles the onChange for input', () => {
  //   expect(changed).toBeFalsy();
  //   const input = instance.findByType('input');
  //   input.simulate('change');
  //   expect(changed).toBeTruthy();
  // });
});
