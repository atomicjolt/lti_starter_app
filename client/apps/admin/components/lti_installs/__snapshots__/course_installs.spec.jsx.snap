// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lti installs course install matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseInstalls
    applicationInstance={
      Object {
        "lti_config_xml": "lti_config_xml",
        "lti_key": "lti_key",
        "lti_secret": "lti_secret",
        "name": "application_name",
      }
    }
    canvasRequest={[Function]}
    courseId={123}
    courseName="courseName"
    courses={Array []}
    installedTool={
      Object {
        "id": 12,
      }
    }
    loadingCourses={
      Object {
        "courseId": "id",
        "courseName": "name",
      }
    }
    onlyShowInstalledChanged={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th>
              <span>
                Course Name
              </span>
            </th>
            <th>
              <div
                className="c-checkbox--right-titlebar"
              />
            </th>
          </tr>
        </thead>,
        <tbody />,
      ],
      "className": "c-table c-table--installs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              <span>
                Course Name
              </span>
            </th>
            <th>
              <div
                className="c-checkbox--right-titlebar"
              />
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                <span>
                  Course Name
                </span>
              </th>,
              <th>
                <div
                  className="c-checkbox--right-titlebar"
                />
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Course Name
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Course Name",
                },
                "ref": null,
                "rendered": "Course Name",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="c-checkbox--right-titlebar"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "c-checkbox--right-titlebar",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                <span>
                  Course Name
                </span>
              </th>
              <th>
                <div
                  className="c-checkbox--right-titlebar"
                />
              </th>
            </tr>
          </thead>,
          <tbody />,
        ],
        "className": "c-table c-table--installs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                <span>
                  Course Name
                </span>
              </th>
              <th>
                <div
                  className="c-checkbox--right-titlebar"
                />
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  <span>
                    Course Name
                  </span>
                </th>,
                <th>
                  <div
                    className="c-checkbox--right-titlebar"
                  />
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Course Name
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Course Name",
                  },
                  "ref": null,
                  "rendered": "Course Name",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="c-checkbox--right-titlebar"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "c-checkbox--right-titlebar",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
